### PR TITLE
feat: Refactor cli.Main into ValidationRunner for easier reuse.

### PR DESCRIPTION
**Pull request by bdferris-v2**
Per discussion in issue #1135 and https://bit.ly/gtfs-validator-packaged-exe#heading=h.fcbgc2jy8dh, we'd like to refactor code in cli.Main into a new class that's easier to call from other contexts, including the new GUI-based app.

This PR moves the bulk of the code in `gtfsvalidator.cli.Main` into a new `gtfsvalidator.runner.ValidationRunner` class that does the work for setting up and running validation.  The configuration for the run is passed via a new ValidationRunnerConfig class.

Both `gtfsvalidator.cli.Main` and `gtfsvalidator.app.gui.Main` have been refactored to use this new validation runner.  The gui app now uses the return code from the runner to potentially open system_errors.json when appropriate.

Command-line argument parsing is left in the existing `Arguments` class, which I've refactored to produce a config object as its primary output.  I'll also merged in the logic from `gtfsvalidator/cli/CliParametersAnalyzer.java`, since it made the interface for `Arguments` simpler and I think conceptually it seems reasonable to include it there.

### CODE REVIEW

**Review by maximearmstrong — CHANGES_REQUESTED** [review]
2 small things before approval. Otherwise, this is great. Thank you @bdferris-v2

File: app/gui/src/main/java/org/mobilitydata/gtfsvalidator/app/gui/Main.java
Comment (bdferris-v2) [reply app/gui/src/main/java/org/mobilitydata/gtfsvalidator/app/gui/Main.java]: Just a quirk of Intellij's import logic.  I inlined it to JOptionPane.

File: main/src/main/java/org/mobilitydata/gtfsvalidator/runner/ValidationRunner.java
Comment (bdferris-v2) [reply main/src/main/java/org/mobilitydata/gtfsvalidator/runner/ValidationRunner.java:183]: Technically still used below in `exportNoticeSchema()`.  I just inlined the method since it's simple and not reused anywhere in the file.

**Comment by bdferris-v2** [comment]
I addressed the suggested changes and test are now fully passing.  I believe this is ready for a final review.

**Comment by maximearmstrong** [comment]
We had to reconfigure the CLA wizard, so if possible, could you re-sign it with your second account? We will be ready to merge right after.

**Comment by bdferris-v2** [comment]
I just resigned.  Let me know if you need anything else?

**Review by maximearmstrong — APPROVED** [review]
All good! Thank you @bdferris-v2

File: main/src/main/java/org/mobilitydata/gtfsvalidator/runner/ValidationRunner.java
Comment (barbeau) [inline main/src/main/java/org/mobilitydata/gtfsvalidator/runner/ValidationRunner.java:125]: The `Status` codes put us in a good position to handle https://github.com/MobilityData/gtfs-validator/issues/1096 as well via something like `Status.PARSING_FAILED`, which is to surface the condition where parsing failed and many validators weren't invoked. That can be in a future PR.

**Review by barbeau — APPROVED** [review]
LGTM, thanks @bdferris-v2!

**Pull request merged**
